lib/beaker-hostgenerator/parser.rb

```diff
@@ -14,7 +14,28 @@ module BeakerHostGenerator
   # each token you would call `is_ostype_token?` and/or `parse_node_info_token`.
   module Parser
 
-    # Capture role and bit width information about the node.
+    # Parses a single node definition into the following components:
+    #
+    #   * bits             Uppercase-only alphanumeric
+    #                      Examples: 32, 64, POWER7 (note 7POWER is invalid)
```

Comment:
It seems unintuitive to me that we would support `POWER7` but not `7POWER`.

This particular pattern is a bit restrictive. We already have a use case that does not fit into it with the `S390X` architecture for zLinux.

I applaud the backward compatible solution, but would prefer to see a regex pattern that grabs everything before the first lower case alpha character as the given architecture.